analytics/terraform/spark-eks-ipv6/addons.tf

```diff
@@ -0,0 +1,571 @@
+            values: ["r4", "r4", "r5", "r5d", "r5n", "r5dn", "r5b", "m4", "m5", "m5n", "m5zn", "m5dn", "m5d", "c4", "c5", "c5n", "c5d"]
+          - key: "kubernetes.io/arch"
+            operator: In
+            values: ["amd64"]
+          - key: "karpenter.k8s.aws/instance-generation"
+            operator: Gt
+            values: ["2"]
+        limits:
+          cpu: 1000
+        disruption:
+          consolidationPolicy: WhenEmpty
+          consolidateAfter: 30s
+          expireAfter: 720h
+        weight: 100
+      EOT
+      ]
+    }
+  }
+}
+
+#---------------------------------------------------------------
+# S3 bucket for Spark Event Logs and Example Data
+#---------------------------------------------------------------
+#tfsec:ignore:*
+module "s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 3.0"
+
+  bucket_prefix = "${local.name}-spark-logs-"
+
+  # For example only - please evaluate for your environment
+  force_destroy = true
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  tags = local.tags
+}
+
+# Creating an s3 bucket prefix. Ensure you copy Spark History event logs under this path to visualize the dags
+resource "aws_s3_object" "this" {
+  bucket       = module.s3_bucket.s3_bucket_id
+  key          = "spark-event-logs/"
+  content_type = "application/x-directory"
+}
+
+#---------------------------------------------------------------
+# Grafana Admin credentials resources
+#---------------------------------------------------------------
+data "aws_secretsmanager_secret_version" "admin_password_version" {
+  secret_id  = aws_secretsmanager_secret.grafana.id
+  depends_on = [aws_secretsmanager_secret_version.grafana]
+}
+
+resource "random_password" "grafana" {
+  length           = 16
+  special          = true
+  override_special = "@_"
+}
+
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "grafana" {
+  name                    = "${local.name}-grafana"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "grafana" {
+  secret_id     = aws_secretsmanager_secret.grafana.id
+  secret_string = random_password.grafana.result
+}
+
+# This is a ConfigMap designed for the Spark Operator version 1.4.6, enabling the exposure of environment variables in an Amazon EKS cluster configured with IPv6.
+# The current Helm values file only supports the 'envFrom' parameter. Starting from the 2.0.0 release, the Helm chart will introduce support for the 'env' parameter, allowing for more flexible configuration options.
+resource "kubernetes_namespace_v1" "spark_operator" {
+  metadata {
+    name = "spark-operator"
+  }
+  timeouts {
+    delete = "15m"
+  }
+}
+
+resource "kubernetes_config_map" "spark_operator_ipv6_configmap" {
```

Comment:
There are some IPv6-specific things like this sprinkled around the solution, should we document those in the docs as well to help show how/why we added things?

Reply:
There is a comment about it above the resource